sympy/assumptions/handlers/matrices.py

```diff
@@ -38,6 +38,8 @@ def MatMul(expr, assumptions):
         factor, mmul = expr.as_coeff_mmul()
         if all(ask(Q.symmetric(arg), assumptions) for arg in mmul.args):
             return True
+        if ask(Q.diagonal(expr), assumptions):
```

Comment:
I think that this should not be necessary. There is a general [rule](https://github.com/sympy/sympy/blob/master/sympy/assumptions/ask.py#L1517) that makes all diagonal matrices symmetric.